src/app/home/home.component.ts

```diff
@@ -42,21 +45,46 @@ export class HomeComponent implements OnInit {
     });
   }
 
+  openSnackBar(message: string, action: string) {
+    this._snackBar.open(message, action, {
+      duration: 7000,
+    });
+  }
+
   getWeather(location: string) {
     this.isLoading = true;
     this.weatherService.get(location)
       .pipe(finalize(() => { this.isLoading = false; }))
-      .subscribe((res: IWeather) => {
-        this.weather = res;
-        this.current = this.weather.current;
-    });
+      .subscribe(
+        (res: IWeather) => {
+          this.weather = res;
+          this.current = this.weather.current;
+        },
+        (err: IError) => {
+          let unhandledError = 'Unhandled error';
+          let errMessage = 
+            err ?
+              err.error ?
+                err.error.error ?
+                  err.error.error.message ?
+                    err.error.error.message
+                  : unhandledError
+                  : unhandledError
+                  : unhandledError
+                  : unhandledError;
```

Comment:
hmmmm I don't like it :stuck_out_tongue: , what if we use lodash? I saw that you already used it in another files :wink:

Reply:
I pushed an alternative but I didn't use lodash or at least I think so. Could you take a look please? Thanks :D

Reply:
This new approach looks better, but if the `error` object is undefined its going to throw an exception, a workaround to avoid it is using the `get` method of lodash: https://lodash.com/docs/4.17.15#get
Lets try to use it!